machine/v1beta1/types_vsphereprovider.go

```diff
@@ -198,8 +198,30 @@ type VSphereDisk struct {
 	// +kubebuilder:validation:Maximum=16384
 	// +required
 	SizeGiB int32 `json:"sizeGiB"`
+	// provisioningMode specifies the provisioning type to be used by this vSphere data disk.
+	// If not set, the setting will be provided by the default storage policy.
```

Comment:
For an enum field, the Godoc generally should follow a format of:
```
provisioningMode specifies the provisioning type to be used by this vSphere data disk.

Allowed values are ...

When set to {Value} ... (repeated for allowed values)
```

Reply:
@everettraven / @JoelSpeed , I can make these changes.  I did not see any existing enum in API doing this.  Do we have this requirement documented anywhere for future enhancers?  I'll be sure to add this requirement to our team's documentation on working with API to reduce missed points.

Reply:
The general guidance for writing user readable godoc is here: https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#write-user-readable-documentation-in-godoc

You don't have to have it match exactly what I put, but that formatting is a good baseline. As long as the Godoc is reasonably structured and answers the key questions in the api conventions doc, it should be OK.

Reply:
>  I did not see any existing enum in API doing this

There are definitely many around that I've API reviewed, if you search for `Allowed values are` in this repo there are many hits

> Do we have this requirement documented anywhere for future enhancers?

Somewhat covered in the general user readable docs section that @everettraven linked, but I think we could likely add more

Reply:
Thank, i'll use that link in my doc as well.

Reply:
@everettraven how do these changed look?